### PR TITLE
 config-dev branch

### PR DESCRIPTION
**Pull request by maringuu**
This branch is intended to transition between the old way of using the config (passing it around) to the new way (importing it).

### CODE REVIEW

File: src/test/unit/analysis/test_plugin_base.py [hunk 1]
```diff
@@ -133,16 +133,27 @@ def tearDown(self):
 
     @mock.patch('plugins.base.ViewUpdater', lambda *_: None)
     def multithread_config_test(self, multithread_flag, threads_in_config, threads_wanted):
-        self.config.set('dummy_plugin_for_testing_only', 'threads', threads_in_config)
-        self.p_base = DummyPlugin(self, self.config, no_multithread=multithread_flag)
-        self.assertEqual(
-            self.p_base.config[self.p_base.NAME]['threads'], threads_wanted, 'number of threads not correct'
-        )
+        self.p_base = DummyPlugin(self, no_multithread=multithread_flag)
+        self.assertEqual(self.p_base.thread_count, int(threads_wanted), 'number of threads not correct')
```
Comment (jstucke) [inline src/test/unit/analysis/test_plugin_base.py:137]: `threads_in_config` seems to be unused

File: src/test/unit/scheduler/test_analysis.py [hunk 2]
```diff
@@ -81,7 +92,7 @@ def test_expected_plugins_are_found(self):
 
         assert 'file_hashes' in result, 'file hashes plugin not found'
         assert 'file_type' in result, 'file type plugin not found'
-        assert 'dummy_plug_in_for_testing_only' not in result, 'dummy plug-in not removed'
+        assert 'dummy_plugin_for_testing_only' not in result, 'dummy plug-in not removed'
```
Comment (jstucke) [inline src/test/unit/scheduler/test_analysis.py:95]: was this test broken all the time? :see_no_evil:

File: src/test/unit/scheduler/test_analysis.py [hunk 3]
```diff
@@ -109,6 +120,20 @@ def test_process_next_analysis_unknown_plugin(self):
...
+                'hashes': 'md5, sha1, sha256, sha512, ripemd160, whirlpool',
+            },
+            'printable_strings': {
+                'min-length': 6,
+            },
+            'dummy_plugin_for_testing_only': {
+                'mime_whitelist': 'foo, bar',
+            },
+        }
+    )
```
Comment (jstucke) [inline src/test/unit/scheduler/test_analysis.py:136]: I don't know if there is a good solution to this
the easiest obvious solution though: saving the common parts in a variable
```suggestion
    # TODO this marker should not completely overwrite the outer markers
    @pytest.mark.cfg_defaults(
        {
            **common_stuff,
            'dummy_plugin_for_testing_only': {
                'mime_whitelist': 'foo, bar',
            },
        }
    )
```

File: src/scheduler/analysis.py [hunk 4]
```diff
@@ -241,6 +242,10 @@ def get_plugin_dict(self) -> dict:
             for plugin_set in plugin_sets:
                 current_plugin_plugin_sets[plugin_set] = plugin in plugin_sets[plugin_set]
             blacklist, whitelist = self._get_blacklist_and_whitelist_from_plugin(plugin)
+            try:
+                thread_count = self.config.get(plugin, "threads", 0)
```
Comment (jstucke) [inline src/scheduler/analysis.py:246]: >   File "/home/jstucke/git/github/FACT_core/src/scheduler/analysis.py", line 246, in get_plugin_dict
>    thread_count = self.config.get(plugin, "threads", 0)
> TypeError: RawConfigParser.get() takes 3 positional arguments but 4 were given

```suggestion
                thread_count = self.config.get(plugin, "threads", fallback=0)
```

File: src/test/unit/scheduler/test_analysis.py (hunk 3 above)
Comment (maringuu) [reply src/test/unit/scheduler/test_analysis.py:136]: The test was to resolve the TODO in conftest.py.
Now the markers are combined.

File: src/scheduler/analysis.py (hunk 4 above)
Comment (maringuu) [reply src/scheduler/analysis.py]: Whoops.

**Comment by maringuu** [comment]
Reminder when config-migration is merged in this branch:

        the version needs to be updated to "2.2.2" when master is merged into this branch and we can probably remove the dependency from requirements_frontend

_Originally posted by @jstucke in https://github.com/fkie-cad/FACT_core/pull/883#discussion_r1025377457_

**Comment by maringuu** [comment]
> Reminder when config-migration is merged in this branch:

Done.

**Comment by maringuu** [comment]
@jstuke This is now ready for merging.
Since both PRs that were merged in this PR were reviewed there should not be much to review here.
I tried to have good commit messages so individual changes should be easy to track.

**Comment by maringuu** [comment]
CI failure is unrelated.

**Comment by jstucke** [comment]
I think a successful CI run would be good nonetheless. Also the last merge seems to have introduced some conflicts :frowning:

**Review by jstucke — APPROVED** [review]
Should be good since the individual changes were reviewed and the CI run was successful

**Pull request merged**